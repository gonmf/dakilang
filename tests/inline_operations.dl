is_power(N, NN) :- eql(0 + (N * N) + ((1 - 3) + 2), NN, Res).

listing

is_power(3, 9)? # Yes

is_power(4, 12)? # No

print_more(N) :- print(N, Out1), print(N * 2, Out2), print(N * N, Out3), print((N + 1) * N + 1, Out4).
print_more(5)?

not(X, Y) :- missing_clause(X, Y).

not(3, 4)?
