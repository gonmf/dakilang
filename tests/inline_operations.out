> is_power(N, NN) :- eql(0 + (N * N) + ((1 - 3) + 2), NN, Res).
>
> listing
0: is_power(N, NN) :- eval(N, '0+($*$)+((1-3)+2)', $1), eql($1, NN, Res).

>
> is_power(3, 9)? # Yes
Yes

>
> is_power(4, 12)? # No
No

>
> print_more(N) :- print(N, Out), print(N * 2, Out), print(N * N, Out), print((N + 1) * N + 1, Out).
> print_more(5)?
5
10
25
31
Yes

>
> not(X, Y) :- eql(X, Y).
>
> not(3, 4)?
No

